master/buildbot/secrets/providers/vault_hvac.py

```diff
@@ -0,0 +1,164 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+"""
+HVAC based providers
+"""
+
+
+from buildbot import config
+from buildbot.secrets.providers.base import SecretProviderBase
+
+import hvac
+
+
+class HvacAuthenticator:
+    """
+    base HVAC authenticator class
+    """
+
+    def authenticate(self, client):
+        pass
+
+
+class HvacAuthenticatorToken(HvacAuthenticator):
+    """
+    HVAC authenticator for static token
+    """
+
+    def __init__(self, token):
+        self.token = token
+
+    def authenticate(self, client):
+        client.token = self.token
+
+
+class HvacAuthenticatorApprole(HvacAuthenticator):
+    """
+    HVAC authenticator for Approle login method
+    """
+
+    def __init__(self, roleId, secretId):
+        self.roleId = roleId
+        self.secretId = secretId
+
+    def authenticate(self, client):
+        client.auth.approle.login(role_id=self.roleId, secret_id=self.secretId)
+
+
+class HvacKvSecretProvider(SecretProviderBase):
```

Comment:
I am not too sure about the name. I would prefer it has Vault in it.  hvac is implementation detail

Reply:
How about `VaultKvSecretProvider`? I have used Hvac in name to distinguish it from old implementation, but you're right this is just implementation detail and should not be presented to user.